api/common/v1alpha1/ref.go

```diff
@@ -38,8 +38,10 @@ var order = map[TargetRefKind]int{
 type TargetRefProxyType string
 
 var (
-	Sidecar TargetRefProxyType = "Sidecar"
-	Gateway TargetRefProxyType = "Gateway"
+	Sidecar     TargetRefProxyType = "Sidecar"
+	Gateway     TargetRefProxyType = "Gateway"
+	ZoneIngress TargetRefProxyType = "ZoneIngress"
+	ZoneEgress  TargetRefProxyType = "ZoneEgress"
```

Comment:
Once we merge this PR, it shouldn't be possible to use the `ZoneIngress` and `ZoneEgress` types in `targetRef.proxyTypes`. Do we need to add validation somewhere? 

Or should we create a new type with 4 values (maybe in a different package) and deprecate `TargetRefProxyType`?

Reply:
right, I've forgot to check usages of this properly, I've added validation to disallow using new types

Reply:
Please create a new type with 4 values, probably in `dataplane_helpers.go` the type is `ProxyTypeLabelValue`